pkg/proto/datadog/trace/agent_payload.proto

```diff
@@ -29,4 +29,6 @@ message AgentPayload {
         // idxTracerPayloads specifies list of the payloads received from tracers.
         // @gotags: msg:"-"
         repeated idx.TracerPayload idxTracerPayloads = 11;
+        // APMMode specifies whether using "full" or other APM modes, it is "full" by default
+        string APMMode = 12;
```

Comment:
I would consider using an enum to represent this field, specially if we want to attach it to every single payload.